### PR TITLE
Update changelog, config docs, JVM recommendation, requirements, and clean up intro documentation.

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

**Comment by ewencp** [comment]
@granders @Ishiihara Can either of you review? Pretty straightforward updates.

File: docs/deployment.rst [hunk 1]
```diff
@@ -73,14 +73,19 @@ latency access to both ZooKeeper and the Kafka brokers. Standard data center net
 JVM
 ~~~
 
-We recommend running JDK 1.7 u51, and using the G1 collector. If you do this (and we highly recommend it), make
+We recommend running the latest version of JDK 1.8 with the G1 collector (older freely available versions have disclosed security vulnerabilities).
```
Comment (Ishiihara) [inline docs/deployment.rst]: Maybe restrict to 100 characters per line?

**Comment by Ishiihara** [comment]
@ewencp It seems that we refer to `confluent-2.1` in the deployment section of `intro.rst` in. We should probably change it to 3.0.

File: docs/deployment.rst (hunk 1 above)
Comment (ewencp) [inline docs/deployment.rst]: Ack. Had to remember how to toggle line length, different files across different projects are using completely different limits....

**Comment by ewencp** [comment]
@Ishiihara Good catch on the version, kinda unfortunate that we don't have an easy way to automate that value -- annoying to have to update for every release. Fixed this up in schema-registry PR as well.

**Comment by Ishiihara** [comment]
LGTM

**Pull request merged**